exonum/src/runtime/dispatcher/mod.rs

```diff
@@ -188,11 +244,15 @@ impl Dispatcher {
         // Currently the only check is that destination service exists, but later
         // functionality of this method can be extended.
         let call_info = &tx.as_ref().call_info;
-        Schema::new(snapshot)
+        let instance = Schema::new(snapshot)
             .get_instance(call_info.instance_id)
             .ok_or(Error::IncorrectInstanceId)?;
 
-        Ok(())
+        if instance.is_active() {
```

Comment:
Are you sure this is the correct filter condition? If I understand correctly, this is *formally* correct (`snapshot` with which `check_tx` is called cannot have pending service statuses set), but logically processing transactions with a pending transition to active state is a mistake.